pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -137,9 +159,10 @@ def advance(self, dataloader_iter: Iterator, **kwargs: Any) -> None:
         processed_batch_end_outputs = self._prepare_outputs(batch_end_outputs, batch_mode=True)
 
         # hook
-        self.trainer.call_hook(
-            "on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, self._dataloader_idx
-        )
+        if not isinstance(self.batch_loop, IteratorBatchProcessor):
+            self.trainer.call_hook(
+                "on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, self._dataloader_idx
+            )
         self.trainer.call_hook("on_batch_end")
         self.trainer.logger_connector.on_batch_end()
 
```

Comment:
(for a separate PR) - this might be pointing out that we need a callback hook for `on_train_step_start` and `on_train_step_end`. the current hooks assume its a batch input and are named accordingly